### PR TITLE
Fix DailyDataGoal not updating when date changes

### PR DESCRIPTION
**Pull request by reneefromhold**
## Overview

When using the DailyDataGoal within a DateRangeCoordinator (already supported), when the user navigates to a new day

**Current Behavior:**
The data does not update
The data is searched for across a span of 3 days

**Expected Behavior:**
The data should reflect the current date
The data is searched for should be limited to one day

## Security

REMINDER: All file contents are public.

- [ ] I have ensured no secure credentials or sensitive information remain in code, metadata, comments, etc.
  - [ ] Please verify that you double checked that .storybook/preview.js does not contain your participant access key details. 
  - [ ] There are no temporary testing changes committed such as API base URLs, access tokens, print/log statements, etc.
- [ ] These changes do not introduce any security risks, or any such risks have been properly mitigated.

Describe briefly what security risks you considered, why they don't apply, or how they've been mitigated.

### Testing
- [ ] MyDataHelps iOS app
- [ ] MyDataHelps Android app
- [ ] [MyDataHelps website](mydatahelps.org)

### Documentation
- [ ] I have added relevant Storybook updates to this PR as well.
- [ ] If this feature requires a developer doc update, tag @CareEvolution/api-docs.

Consider "Squash and merge" as needed to keep the commit history reasonable on `main`.

## Reviewers

Assign to the appropriate reviewer(s). Minimally, a second set of eyes is needed ensure no non-public information is published. Consider also including:
- Subject-matter experts
- Style/editing reviewers
- Others requested by the content owner

## Summary by CodeRabbit

- **New Features**
	- Introduced a new story for `LiveGarminSleepScore` with updated goal settings and messages.
	- Added a new story for `LiveSleepMinutesWithDateRangeCoordinator` to enhance component rendering.

- **Improvements**
	- Enhanced date handling logic for better accuracy in daily data queries.

- **Refactor**
	- Renamed existing story from `SleepGoalLive` to `LiveSleepMinutes` and updated rendering method for consistency.
	- Removed the `WearTimeGoalLive` story to streamline options.

### CODE REVIEW

File: src/components/container/DailyDataGoal/DailyDataGoal.tsx [hunk 1]
```diff
@@ -33,13 +33,13 @@ export default function (props: DailyDataGoalProps) {
     const date = props.date ?? dateRangeContext?.intervalStart ?? startOfDay(new Date());
 
     useInitializeView(() => {
-        const startDate = add(date, { days: -1 });
-        const endDate = add(date, { days: 2 });
+        const startDate = startOfDay(date);
+        const endDate = endOfDay(date);
```
Comment (ajmenca) [inline src/components/container/DailyDataGoal/DailyDataGoal.tsx:37]: I'm curious - was this actually causing misbehaviors?  I'm pretty sure that, somewhere a couple of layers below this in the actual queries we already "pad" the date range requested for a variety of timezone/sync related reasons, but this code still only looks at the result of a single day.  So was the only "bad effect" of this that we're getting a larger data set back than necessary?

File: src/components/container/DailyDataGoal/DailyDataGoal.tsx [hunk 2]
```diff
@@ -33,13 +33,13 @@ export default function (props: DailyDataGoalProps) {
...
 
         queryDailyData(props.dailyDataType, startDate, endDate, props.previewState === "Default").then((results) => {
             setDailyValue(results[getDayKey(date)] || 0);
         });
-    }, ["externalAccountSyncComplete"], []);
+    }, ["externalAccountSyncComplete"], [dateRangeContext?.intervalStart]);
```
[the remark this answers is not in the document]
Comment (ajmenca) [reply src/components/container/DailyDataGoal/DailyDataGoal.tsx:42]: This is incorrect, as this is not a dependency array being passed to `useEffect`, but an array being passed to `useInitializeView`.  There are purposefully two array parameters to this function, the first is used for registering for MyDataHelps events, and the second is used for `useEffect` dependencies.

File: src/components/container/DailyDataGoal/DailyDataGoal.stories.tsx [hunk 3]
```diff
@@ -114,34 +115,40 @@ export const SleepGoalLive = {
...
 };
 
-export const WearTimeGoalLive = {
+const renderWithDRC = (args: DailyDataGoalProps) => <Layout colorScheme='auto'>
+    <DateRangeCoordinator intervalType="Day">
+        <DailyDataGoal {...args} />
+    </DateRangeCoordinator>
+</Layout>;
+
+export const LiveGarminSleepScore = {
```
Comment (ajmenca) [inline src/components/container/DailyDataGoal/DailyDataGoal.stories.tsx:124]: Minor suggestion - I think the story names are a little confusing here.  It sounds like the only difference between the story you renamed and the story you edited is the type of data, but one also uses the `DateRangeCoordinator` and the other doesn't.  My suggestion would actually be 4 stories in this case; two for each data type with and without the `DateRangeCoordinator`.

File: src/components/container/DailyDataGoal/DailyDataGoal.tsx (hunk 1 above)
Comment (reneefromhold) [reply src/components/container/DailyDataGoal/DailyDataGoal.tsx:37]: This is intended for looking at one day of data, what is the value of padding extra days back/fwd? Is there a use case or something with the returned data that isn't apparent here?

File: src/components/container/DailyDataGoal/DailyDataGoal.stories.tsx (hunk 3 above)
Comment (reneefromhold) [reply src/components/container/DailyDataGoal/DailyDataGoal.stories.tsx:124]: There is a unified sleep dailyDataType, but in this story it wasn't used. I think it makes more sense to update to use the unified vs manf specific sleep data type and revert the name back to imply it is not manf specific data. 

I'll update the new story I added with the Date Range Coordinator to use the same unified data type. There are 65ish DailyDataTypes. Adding a live story for each one is out of scope for this change.

File: src/components/container/DailyDataGoal/DailyDataGoal.tsx (hunk 1 above)
Comment (ajmenca) [reply src/components/container/DailyDataGoal/DailyDataGoal.tsx:37]: I wasn't suggesting changing it back, I was asking if it was *actually* causing a participant facing bad behavior, or just extra data load.  As I said, in the underlying code it buffers the timestamps slightly to ensure you actually get the data you're looking for, so this double buffering certainly isn't necessary.  I was just curious.

Comment (reneefromhold) [reply src/components/container/DailyDataGoal/DailyDataGoal.tsx:37]: No, it wasn't showing the wrong data. I came across it while troubleshooting and found it misleading

**Review by ajmenca — APPROVED** [review]

**Pull request merged**